### ISSUE TITLE
Duplicate server header

### ISSUE THREAD

**Issue by utrenkner** [h2o/h2o#2150]
My h2o v 2.3.0-beta2 sends out duplicate server headers.
```
content-length: 126110
content-type: text/html; charset=UTF-8
date: Sat, 12 Oct 2019 14:53:35 GMT
server: h2o/2.3.0-beta2
server: h2o/2.3.0-beta2
status: 200
strict-transport-security: max-age=31536000
x-content-type-options: nosniff
x-xss-protection: 1; mode=block
```
I did not yet find the time to dive deeper into the issue (it must be either related to reverse-proxying or similar to the duplicate date header, which [stemmed from mruby](https://github.com/h2o/h2o/pull/1780)).

I tried to overwrite the header with ``header.set: "server: h2o/2.3.0-beta2"`` and also a combination of ``send-server-name: Off`` and ``header.add: "server: h2o/2.3.0-beta2"``. The answer was always the same ``in command header.add, the named header cannot be rewritten``. Likewise for ``header.unset``. It looks to me as if the server-header is protected.

**Comment by kazuho** [comment]
How are you generating the response? Is it by using mruby to access the origin server?

**Comment by utrenkner** [comment]
Yes, in the cases, where the duplicate server header appears, the response comes from mruby using ``http_request()`` to the origin server.

**Comment by kazuho** [comment]
Thank you for checking that. I see that there is a bug.

H2O is the origin server and therefore it should be dropping the server header that it received from the backend. IMO that should be done within the mruby handler, after the mruby code gives h2o the response to be sent. However, that logic seems to be lacking.

In the meantime, as a workaround, I think you can drop the server header when you process the output of `http_request()`.

**Comment by utrenkner** [comment]
Thank you for the workaround, @kazuho !

For reference - if someone stumbles on this issue before it is fixed. I just added a ``h.delete('server')`` in my mruby code and the duplicate server header disappears:

```
headers = {}
headers["my-additional-header"] = "my-additional-header-value"
req = http_request("#{server}#{path}",{
   headers: headers,
})
s, h, b = req.join
h.delete('server')
return [ s, h, b]
```
